packages/manager/cypress/e2e/core/cloudpulse/dbaas-widgets-verification.spec.ts

```diff
@@ -101,28 +102,50 @@ const metricsAPIResponsePayload = cloudPulseMetricsResponseFactory.build({
 });
 
 /**
- * Verifies the presence and values of specific properties within the aclpPreference object
- * of the request payload. This function checks that the expected properties exist
- * and have the expected values, allowing for validation of user preferences in the application.
+ * Generates graph data from a given CloudPulse metrics response and
+ * extracts average, last, and maximum metric values from the first
+ * legend row. The values are rounded to two decimal places for
+ * better readability.
+ *
+ * @param responsePayload - The metrics response object containing
+ *                          the necessary data for graph generation.
+ * @param label - The label for the graph, used for display purposes.
+ *
+ * @returns An object containing rounded values for max average, last,
  *
- * @param requestPayload - The payload received from the request, containing the aclpPreference object.
- * @param expectedValues - An object containing the expected values for properties to validate against the requestPayload.
- *    Expected properties may include:
- *    - dashboardId: The ID of the dashboard.
- *    - timeDuration: The selected time duration for metrics.
- *    - engine: The database engine used.
- *    - region: The selected region for the dashboard.
- *    - resources: An array of resource identifiers.
- *    - role: The role associated with the dashboard user.
  */
+
 const getWidgetLegendRowValuesFromResponse = (
-  responsePayload: CloudPulseMetricsResponse
+  responsePayload: CloudPulseMetricsResponse,
+  label: string,
+  unit: string
 ) => {
-  const data = transformData(responsePayload.data.result[0].values, 'Bytes');
-  const { average, last, max } = getMetrics(data);
-  const roundedAverage = Math.round(average * 100) / 100;
-  const roundedLast = Math.round(last * 100) / 100;
-  const roundedMax = Math.round(max * 100) / 100;
+  // Generate graph data using the provided parameters
+  const graphData = generateGraphData({
+    flags: { enabled: true } as Partial<Flags>,
+    label: label,
+    metricsList: responsePayload,
+    resources: [
+      {
+        id: '1',
+        label: clusterName,
+        region: 'us-ord',
+      },
+    ],
+    serviceType: serviceType,
+    status: 'success',
+    unit: unit,
+    // widgetColor: 'red',
```

Comment:
Do we need this commented code now?

Reply:
@agorthi-akamai  please remove this comment, as for now we are not taking care of widget color coming from configuration